extensions/wikia/AdEngine/js/template/bfaaDesktop.js

```diff
@@ -72,20 +71,37 @@ define('ext.wikia.adEngine.template.bfaaDesktop', [
 			recoveryTweaker.tweakSlot(params.slotName, iframe);
 		}
 
-		if (params.videoUrl && params.videoTriggerElement) {
-			slotTweaker.onReady(params.slotName, function() {
-				var divs = doc.querySelectorAll('#' + params.slotName + ' > div'),
-					imageContainer = divs[divs.length - 1],
-					video = uapVideoAd.init(doc.getElementById(params.slotName), imageContainer, params.videoUrl);
-
-				params.videoTriggerElement.addEventListener('click', function () {
-					uapVideoAd.playAndToggle(video, imageContainer);
-				});
+		if (params.videoTriggerElement && params.videoAspectRatio) {
+			videoAdFactory.init().then(function () {
+				try {
+					var video = videoAdFactory.create(
+						adSlot.querySelector('div:last-of-type'),
```

Comment:
Parameters passed here are extremely similar to [the ones in mobile version](https://github.com/Wikia/app/pull/11744/files#diff-0a5b987183cc7a5dcc3807570e8476adR83). What do you think of creating default ones and then just updating/extending them?